apps/cli/lib/mix/tasks/mana.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Mix.Tasks.Mana do
+  @moduledoc """
+  Main entry-point for the mana CLI.
+
+  Command Line Options:
+
+      `--chain` - Chain to load data from (default: ropsten)
+      `--discovery` - Perform discovery (default: true)
+      `--sync` - Perform syncing (default: true)
+      `--bootnodes` - Comma separated list of bootnodes (default from chain)
+
+  Examples:
+
+      # Sync from a local node
+
+      mix mana --chain ropsten --discovery false --bootnodes enode://...
+  """
+  use Mix.Task
+  require Logger
+
+  @shortdoc "Starts Mana application"
+  def run(args) do
+    case CLI.Parser.ManaParser.mana_args(args) do
+      {:ok,
+       %{
+         chain_name: chain_name,
+         discovery: discovery,
+         sync: sync,
+         bootnodes: bootnodes
+       }} ->
+        :ok = Logger.warn("Starting mana chain #{Atom.to_string(chain_name)}...")
+
+        ExWire.Config.configure!(
+          chain: chain_name,
+          sync: sync,
+          discovery: discovery,
+          bootnodes: bootnodes
+        )
+
+        Application.ensure_all_started(:ex_wire)
+
+        # No Halt
+        Process.sleep(:infinity)
```

Comment:
For what purpose `Process.sleep` is used here?

from its docs:
```
Use this function with extreme care. 
For almost all situations where you would use sleep/1 in Elixir, there is likely a more correct, faster and precise way of achieving the same with message passing.
```

Reply:
It's just so the process doesn't exit since it otherwise finishes the only function it runs (but it started other supervised applications, so it still is being productive). I took this line from here: https://github.com/elixir-lang/elixir/blob/master/lib/mix/lib/mix/tasks/run.ex#L87

Reply:
Not sure if we could do something like Phoenix does when it starts a [web server], defaulting the "--no-halt" flag. Not sure what modifications we'd have to do in order to do that, but just throwing another option out there (although it seems from what you linked to, that's what --no-halt does underneath the hood).

Reply:
We would probably need to add something like a deamon flag as well...